packages/cli/src/external-secrets.ee/types.ts

```diff
@@ -0,0 +1,56 @@
+import type { IDataObject, INodeProperties } from 'n8n-workflow';
+
+import type { AuthenticatedRequest } from '@/requests';
+
+export interface SecretsProviderSettings<T = IDataObject> {
+	connected: boolean;
+	connectedAt: Date | null;
+	settings: T;
+}
+
+export interface ExternalSecretsSettings {
+	[key: string]: SecretsProviderSettings;
+}
+
+export type SecretsProviderState = 'initializing' | 'connected' | 'error';
+
+export abstract class SecretsProvider {
+	displayName: string;
+
+	name: string;
+
+	properties: INodeProperties[];
+
+	state: SecretsProviderState;
+
+	abstract init(settings: SecretsProviderSettings): Promise<void>;
+	abstract connect(): Promise<void>;
+	abstract disconnect(): Promise<void>;
+	abstract update(): Promise<void>;
+	abstract test(): Promise<[boolean] | [boolean, string]>;
+	abstract getSecret(name: string): unknown;
+	abstract hasSecret(name: string): boolean;
+	abstract getSecretNames(): string[];
+}
+
+export declare namespace ExternalSecretsRequest {
```

Comment:
Does this belong in `@n8n/api-types`?

Reply:
There will be a follow up PR to setup DTOs for all these endpoints. I already started that on a branch.